### PR TITLE
Refactoring and clean-up group API

### PR DESCRIPTION
**Pull request by dmitry-ganyushin**

### CODE REVIEW

**Comment by eisenhauer** [comment]
Can you summarize these changes?  (And maybe rebase on current release_29 so it's not out-of-date?)

**Comment by eisenhauer** [comment]
Still curious about what's driving this.  This includes changes to the external API as well as internal implementation. The affected routines don't seem to be documented in the user guide, but what is driving their removal?  Is this preparation for some other more extensive work?  Or a more casual decision to change stuff?

**Review by eisenhauer — CHANGES_REQUESTED** [review]
Just starting a review, requests for details/motivation are in other comments.

**Comment by dmitry-ganyushin** [comment]
Was a quick initial string-based implementation. Contained some debug functions and loosely access to internal functions. Was just a bit long in the TODO list for finalizing.

**Comment by eisenhauer** [comment]
So, this is cleanup from years ago?  Certainly there does seem to be some weirdness in there.   You want to put off merging this until after the release in order to avoid complicating life for @vicentebolea ?

**Review by pnorbert — APPROVED** [review]
This is a cleanup and can go into the release.

**Comment by dmitry-ganyushin** [comment]
It is not used currently so no damage can be possibly done.

**Comment by eisenhauer** [comment]
> It is not used currently so no damage can be possibly done.

It's been a public API for years, so asserting that it isn't used is hard.  But I'm not too worried about the consequences...

**Review by eisenhauer — APPROVED** [review]
